src/audio/kpb.c

```diff
@@ -965,7 +967,7 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 		/* In draining intervals we fill only half of host buffer.
 		 * This was we are safe to not overflow it.
 		 */
-		period_bytes_limit = host_buffer_size / 2;
+		period_bytes_limit = host_period_size * 2;
```

Comment:
Comment should be changed also?

Reply:
```
"Instead of copying half the buffer each period we
copy two periods. This slightly decreases draining
speed but avoids XRUNs in rare cases."
```
Hey, here the "host_period_size * 2" is not necessary smaller than "host_buffer_size / 2", imagine there are only 2 periods in the host_buffer.

Reply:
That's right, `host_period_size * 2`  maybe equal to half the buffer size but the point is it might as well be 1/8 or even smaller fraction of the host buffer. So in order to limit the likelihood of XRUNs we really shouldn't copy more than 2 periods at once.